### PR TITLE
pcq-114-pit-testing-configuration

### PR DESCRIPTION
**Pull request by ManojKulkarni-HMCTS**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PCQ-114

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -7,6 +7,7 @@ plugins {
   id 'org.springframework.boot' version '2.2.5.RELEASE'
   id 'org.owasp.dependencycheck' version '5.3.0'
   id 'com.github.ben-manes.versions' version '0.28.0'
+  id "info.solidsoft.pitest" version '1.3.0'
```
Comment (davejones74) [inline build.gradle]: Could this be upgraded to 1.5.0 and same for the relevent plugins / gradlepitest

File: build.gradle [hunk 2]
```diff
@@ -133,11 +134,32 @@ jacocoTestReport {
 
 project.tasks['sonarqube'].dependsOn jacocoTestReport
 
+pitest {
+  targetClasses = ['uk.gov.hmcts.reform.pcqbackend.*']
+  excludedClasses = [
+    'uk.gov.hmcts.reform.pcqbackend.configuration.*',
+    'uk.gov.hmcts.reform.pcqbackend.controller.*',
```
Comment (davejones74) [inline build.gradle]: Any reason why controller is excluded? Shouldn't be necessary?

File: build.gradle [hunk 3]
```diff
@@ -133,11 +134,32 @@ jacocoTestReport {
...
+    'uk.gov.hmcts.reform.pcqbackend.Application'
+  ]
+  threads = 10
+  enableDefaultIncrementalAnalysis = true
+  historyInputLocation = ['build/reports/pitest/fastermutationtesting']
+  historyOutputLocation = ['build/reports/pitest/fastermutationtestingoutput']
+  outputFormats = ['XML', 'HTML']
+  timestampedReports = true
+  mutationThreshold = 90
+
```
Comment (davejones74) [inline build.gradle]: Just a note probate has the following differences
  mutationThreshold = 50
  timeoutConstInMillis = 30000

Do we need a timeout?

File: build.gradle (hunk 1 above)
Comment (ManojKulkarni-HMCTS) [reply build.gradle]: Ok, can do it.

File: build.gradle (hunk 2 above)
Comment (ManojKulkarni-HMCTS) [reply build.gradle]: Nothing much in the controller in terms of business logic and hence excluded.

File: build.gradle (hunk 3 above)
Comment (ManojKulkarni-HMCTS) [reply build.gradle]: I took this setting from another project that Satya recommended but can add the settings as per probate. Not sure about the timeout. What is the timeout for?

File: build.gradle (hunk 1 above)
Comment (ManojKulkarni-HMCTS) [reply build.gradle]: 1.5.0 is giving errors that it cannot be found. I checked on the website and the latest version is 1.4.7 so I will configure that one.

Comment (ManojKulkarni-HMCTS) [reply build.gradle]: 1.5.0 is not available. The latest version is 1.4.7 but it is complaining about the gradle version. I have taken these settings from a project that was recommended to me by the Platform engineering guy. I will use this for now as it is working fine.

File: build.gradle (hunk 3 above)
Comment (ManojKulkarni-HMCTS) [reply build.gradle]: Have added the probate settings now.

**Pull request merged**